### PR TITLE
nx - issue#7 - filter images with tags

### LINKED ISSUE #7: Images by Tag
#### As a user I want to view all images associated with a tag.

__Story__:

Now that we have added the ability for images to be associated with tags, let
us provide the ability for users to filter images by a given tag. Nana, who
loves fast planes, will be delighted to see only images labeled with SR-71.

__Acceptance criteria__:
- [x] When I click on a tag, I see a filtered list of only the images that have
  the tag I clicked on.
- [x] The tag name, not its ID, is used to perform the filtering.

__Discussion topics__:
- [x] Where in the code should we implement this functionality?
- [x] What happens if a user tries to manually filter by a nonexistent tag?

__Dependencies__:
- Image Tags

### PR DESCRIPTION
**Pull request by xuinleaf849**
Closes #7

### CODE REVIEW

**Comment by xuinleaf849** [comment]
[[ROPES#7] Display and filter images by Tag](https://trello.com/c/AeRn4gDf/674-ropes7-display-and-filter-images-by-tag)

File: test/controllers/images_controller_test.rb [hunk 1]
```diff
@@ -1,27 +1,42 @@
...
+      { imagelink: 'https://learn.appfolio.com/apm/www/images/apm-logo-v2.png', tag_list: %w[appfolio logo] },
+      { imagelink: 'https://pbs.twimg.com/profile_images/999418366858149888/zKQCw1Ok_400x400.jpg' },
+      { imagelink: 'https://microsmallcap.com/wp-content/uploads/sites/2/2018/01/AppFolio-The-Dip-is-a-Buying-Opportunity-.png', tag_list: ['appfolio'] }  # rubocop:disable Metrics/LineLength
+    ]
+    Image.create(images)
+
+    get images_path
+    assert_response :ok
+
+    assert_select 'li:last-child .tag-class', count: 2
```
[the remark this answers is not in the document]
Comment (xuinleaf849) [reply test/controllers/images_controller_test.rb]: Good catch. Thanks! Added.

File: app/views/images/index.html.erb [hunk 2]
```diff
@@ -1,8 +1,26 @@
...
+    <p></p>
+<% end %>
+
 <ul>
   <% @images.each do |image| %>
       <li><%= image_tag(image.imagelink, width: '400px') %>
+        <% if image.tag_list.present? %>
+
+            <% image.tag_list.each do |tag| %>
+                <div class='tag-class'> <a href="<%= images_path %>?tag=<%= tag %>"> <%= tag %></div>
```
Comment (mosessliu) [inline app/views/images/index.html.erb]: are you manually constructing the `href` here?

Comment (xuinleaf849) [reply app/views/images/index.html.erb]: Thanks for the suggestion! I changed it with `link_to tag, images_path(tag: tag)` and it works!

**Pull request merged**